### DISCUSSION TITLE
Consideration for supply chain vulnerability injection.

### DISCUSSION THREAD

**Discussion by Wind010** [OWASP/wrongsecrets#1310] [General]
Are the sources for the binaries under [/src](https://github.com/OWASP/wrongsecrets/tree/master/src/main/resources/executables) checked in somewhere in a private repository or are the binaries just checked in directly by contributors?  I understand that a build pipeline for each of the architectures is a pain which is why I suspect the binaries are just checked in.  Are the binaries scanned for any malicious code prior to being merged to main repository or just reviewed manually?

I bring this up in recent supply 10.0 chain attack with [CVE-2024-3094](https://nvd.nist.gov/vuln/detail/CVE-2024-3094).

https://www.cisa.gov/news-events/alerts/2024/03/29/reported-supply-chain-compromise-affecting-xz-utils-data-compression-library-cve-2024-3094

Risks:
- Malicious code running in hosted in docker container.
- Users running the malicious code outside a VM or Container.

**Comment by commjoen** [comment]
Thank you for starting a discussion on the matter, as this could have been a big issue indeed. The binaries are build in another [repository](https://github.com/OWASP/wrongsecrets-binaries) ;-) and build locally by me before checking them in. Frameworks and third party libraries used are scanned and managed by Dependabot :-).
So I don’t think I am at risk there :-).
